inst/recorder/app.R

```diff
@@ -331,6 +344,20 @@ shinyApp(
           value = if (load_mode) "myloadtest" else "mytest"),
         checkboxInput("editSaveFile", "Open script in editor on exit", value = TRUE),
         if (!load_mode) checkboxInput("runScript", "Run test script on exit", value = TRUE),
+        checkboxInput(
+          "allowInputNoBinding",
+          tagList("Record inputs that do not have a binding",
+            tooltip(
+              paste(
+                "This enables recording inputs that do not have a binding, which is common in htmlwidgets",
+                "like DT and plotly. Note that playback support is limited: shinytest will set the input",
+                "value so that R gets the input value, but the htmlwidget itself will not be aware of the value."
+              ),
+              placement = "bottom"
+            )
+          ),
+          value = FALSE
+        ),
```

Comment:
Is there any real danger in having this default to `TRUE`? After running your example I was confused as to why the code generator wasn't working, then I realized I had to check this box...

Reply:
I defaulted to `FALSE` because I didn't want people to start recording tests that use this feature, and then be surprised when the testing doesn't work perfectly -- screenshots that don't look right and/or behavior that isn't correct (due to an htmlwidget that needs actual interaction). By requiring users to opt-in, they're more likely to be aware of that.

I do get that the current situation is confusing. I can think of a few different alternatives:

* In the right panel, input events  without a binding could be highlighted in red. The drawback is that this isn't helpful for people who encounter it for the first time -- there needs to be some way to discover how to deal with it. Maybe a question mark and tooltip?
* When the user clicks "Save and Quit", it could pop a confirmation dialog if there are any inputs without a binding and the box is unchecked. The dialog could say something roughly like, "There are input events without a binding. If you want them to be saved in the test script, press Cancel, then check the box, then click on Save and Quit again. If you don't want to save them, press OK."
* By default, it could just generate a script with the code, but have it commented out. Then users could fix it by uncommenting the code.

Reply:
I finally got around to testing this out.  I'd recommend the second alternative (confirmation dialog) if feasible.